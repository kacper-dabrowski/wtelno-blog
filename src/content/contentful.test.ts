import {
  getAllTabs,
  getPageBySlug,
  getPostPreviews,
  getPosts,
} from "./contentful";

describe("contentful integration", () => {
  it("should get page content by slug", async () => {
    expect(await getPageBySlug("regulamin-cmentarza")).toMatchSnapshot();
  });

  it("should get all tabs", async () => {
    expect(await getAllTabs()).toMatchSnapshot();
  });

  it("should return posts", async () => {
    expect(await getPosts()).toEqual("");
  });

  it("should return post previews", async () => {
    expect(await getPostPreviews()).toMatchSnapshot();
  });
});
