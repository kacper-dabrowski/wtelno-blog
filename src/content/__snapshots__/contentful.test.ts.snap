// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contentful integration should get all tabs 1`] = `
[
  {
    "hasSubpages": false,
    "pathname": "regulamin-cmentarza",
    "title": "Regulamin cmentarza",
  },
]
`;

exports[`contentful integration should get page content by slug 1`] = `
{
  "content": {
    "content": [
      {
        "content": [
          {
            "data": {},
            "marks": [],
            "nodeType": "text",
            "value": "Cmentarz jest miejscem świętym, miejscem wiecznego spoczynku naszych Zmarłych i miejscem modlitwy za nich. Ma charakter sakralny. Powinien być otoczony religijną czcią i szacunkiem oraz szczególną troską.",
          },
        ],
        "data": {},
        "nodeType": "paragraph",
      },
      {
        "content": [
          {
            "data": {},
            "marks": [],
            "nodeType": "text",
            "value": "§ 1 Przepisy ogólne",
          },
        ],
        "data": {},
        "nodeType": "heading-2",
      },
      {
        "content": [
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Cmentarz jest własnością ",
                  },
                  {
                    "data": {},
                    "marks": [
                      {
                        "type": "bold",
                      },
                    ],
                    "nodeType": "text",
                    "value": "Parafii pw. Św. Michała Archanioła we Wtelnie",
                  },
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": ".",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Administratorem cmentarza jest ",
                  },
                  {
                    "data": {},
                    "marks": [
                      {
                        "type": "bold",
                      },
                    ],
                    "nodeType": "text",
                    "value": "Proboszcz Parafii",
                  },
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": ".",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Wszelkie sprawy związane z cmentarzem należy załatwiać w biurze parafialnym.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
        ],
        "data": {},
        "nodeType": "ordered-list",
      },
      {
        "content": [
          {
            "data": {},
            "marks": [],
            "nodeType": "text",
            "value": "§ 2. Postanowienia porządkowe",
          },
        ],
        "data": {},
        "nodeType": "heading-2",
      },
      {
        "content": [
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Dysponent prawa do grobu oraz miejsca objętego rezerwacją na miejsce grzebalne, ma obowiązek utrzymania w czystości i porządku tych miejsc, jak również ich otoczenia.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Zwiędłe kwiaty, wieńce, stare znicze i inne nieczystości należy składać do pojemników w wyznaczonym miejscu.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Zarząd cmentarza nie odpowiada za szkody na grobach powstałe na skutek klęsk żywiołowych, czynników atmosferycznych, dzikich zwierząt, kradzieży i aktów wandalizmu oraz za rzeczy pozostawione bez nadzoru, a także szkody wynikające z zaniedbania lub zaniechania. Groby podlegają indywidualnemu ubezpieczeniu.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Każdy grób powinien być czytelnie oznakowany, tj. posiadać krzyż i umieszczoną w widocznym miejscu informację z imieniem i nazwiskiem osoby zmarłej oraz datą jej śmierci.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Ustawianie nagrobków i każda ich zmiana odbywa się po uprzednim zatwierdzeniu projektu w biurze parafialnym. Stare nagrobki należy usunąć z cmentarza na koszt stawiającego nowy nagrobek.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Nagrobki nie mogą przekraczać ustalonych granic grobu i dopuszcza się stawianie pomników o maks. wymiarach: szer. 0,8 m, dł. 1,8 m (pojedynczy); szer. 1,6 m, dł. 1,8 m (podwójny); szer. 0,6 m, dł. 0,9 m (dziecięcy).",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Kopanie grobów ekshumacyjne i obsługę pogrzebu wykonują zakłady pogrzebowe wybrane przez zainteresowanych.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Pozostały po remoncie lub rozbiórce gruz ze starego pomnika po powstaniu nowego nagrobka należy wywieść z cmentarza.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Na cmentarzu zabrania się:",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
              {
                "content": [
                  {
                    "content": [
                      {
                        "content": [
                          {
                            "data": {},
                            "marks": [],
                            "nodeType": "text",
                            "value": "Zakłócania ciszy, porządku i powagi miejsca",
                          },
                        ],
                        "data": {},
                        "nodeType": "paragraph",
                      },
                    ],
                    "data": {},
                    "nodeType": "list-item",
                  },
                  {
                    "content": [
                      {
                        "content": [
                          {
                            "data": {},
                            "marks": [],
                            "nodeType": "text",
                            "value": "Zaśmiecania terenu, niszczenia zieleni, nagrobków i urządzeń cmentarnych",
                          },
                        ],
                        "data": {},
                        "nodeType": "paragraph",
                      },
                    ],
                    "data": {},
                    "nodeType": "list-item",
                  },
                  {
                    "content": [
                      {
                        "content": [
                          {
                            "data": {},
                            "marks": [],
                            "nodeType": "text",
                            "value": "Bez zgody administratora sadzenia i usuwania drzew, krzewów, ustawiania ławek i układania kostki brukowej wokół pomników",
                          },
                        ],
                        "data": {},
                        "nodeType": "paragraph",
                      },
                    ],
                    "data": {},
                    "nodeType": "list-item",
                  },
                  {
                    "content": [
                      {
                        "content": [
                          {
                            "data": {},
                            "marks": [],
                            "nodeType": "text",
                            "value": "Samodzielnego wykonywania prac budowlanych i kamieniarskich",
                          },
                        ],
                        "data": {},
                        "nodeType": "paragraph",
                      },
                    ],
                    "data": {},
                    "nodeType": "list-item",
                  },
                  {
                    "content": [
                      {
                        "content": [
                          {
                            "data": {},
                            "marks": [],
                            "nodeType": "text",
                            "value": "Wrzucania gruzu i odpadów po robotach kamieniarsko-budowlanych do pojemników na śmieci",
                          },
                        ],
                        "data": {},
                        "nodeType": "paragraph",
                      },
                    ],
                    "data": {},
                    "nodeType": "list-item",
                  },
                  {
                    "content": [
                      {
                        "content": [
                          {
                            "data": {},
                            "marks": [],
                            "nodeType": "text",
                            "value": "Wprowadzania psów",
                          },
                        ],
                        "data": {},
                        "nodeType": "paragraph",
                      },
                    ],
                    "data": {},
                    "nodeType": "list-item",
                  },
                  {
                    "content": [
                      {
                        "content": [
                          {
                            "data": {},
                            "marks": [],
                            "nodeType": "text",
                            "value": "Palenia tytoniu i picia alkoholu",
                          },
                        ],
                        "data": {},
                        "nodeType": "paragraph",
                      },
                    ],
                    "data": {},
                    "nodeType": "list-item",
                  },
                  {
                    "content": [
                      {
                        "content": [
                          {
                            "data": {},
                            "marks": [],
                            "nodeType": "text",
                            "value": "Prowadzenia działalności handlowej",
                          },
                        ],
                        "data": {},
                        "nodeType": "paragraph",
                      },
                    ],
                    "data": {},
                    "nodeType": "list-item",
                  },
                ],
                "data": {},
                "nodeType": "unordered-list",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Wykonawcy usług pogrzebowych, kamieniarskich, budowlanych ponoszą wyłączną odpowiedzialność za działania zaniechania swoich pracowników realizujących czyności na terenie cmentarza.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
        ],
        "data": {},
        "nodeType": "ordered-list",
      },
      {
        "content": [
          {
            "data": {},
            "marks": [],
            "nodeType": "text",
            "value": "§ 3. Pochówek i opłaty",
          },
        ],
        "data": {},
        "nodeType": "heading-2",
      },
      {
        "content": [
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Opłatę za grób (tzw. pokładne) uiszcza się przy pochówku za cały 20-letni okres.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Grób nie może być użyty do ponownego chowania przed upływem 20 lat.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "W przypadku grobu głębinowego może nastąpić „dochowanie” w ciągu 20 lat.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Przed upływem 20 lat można do grobu „dochować” urnę.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Po upływie 20 lat należy uiścić opłatę, przedłużając prawo do grobu na następne 10,15,lub 20 lat. Po tym okresie grób zostanie przekazany innej osobie celem pochowania innego zmarłego.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
          {
            "content": [
              {
                "content": [
                  {
                    "data": {},
                    "marks": [],
                    "nodeType": "text",
                    "value": "Istnieje możliwość rezerwacji miejsca na cmentarzu na zasadach takich jak w przypadku pokładnego (opłata na 20 lat). Przy pochówku osoby, która ma miejsce zarezerwowane, pobiera się pokładne na 20 lat, odliczając kwotę należną za lata, które jeszcze nie upłynęły, a już zostały opłacone przy rezerwacji miejsca.",
                  },
                ],
                "data": {},
                "nodeType": "paragraph",
              },
            ],
            "data": {},
            "nodeType": "list-item",
          },
        ],
        "data": {},
        "nodeType": "ordered-list",
      },
      {
        "content": [
          {
            "data": {},
            "marks": [
              {
                "type": "bold",
              },
            ],
            "nodeType": "text",
            "value": "W sprawach nieuregulowanych regulaminem mają zastosowanie powszechnie obowiązujące przepisy, a w szczególności Ustaw z dnia 31 stycznia 1956 r. o cmentarzach i chowaniu zmarłych (Dz.U. z 2000 r. Nr 23, poz. 295 z późn. zm.) oraz Rozporządzenie Ministra Infrastruktury z dnia 7 marca 2008 r. w sprawie wymagań, jakie muszą spełniać cmentarze, groby i inne miejsca pochówku zwłok i szczątków (Dz.U. z 2008 r. Nr 48, poz. 284).",
          },
        ],
        "data": {},
        "nodeType": "paragraph",
      },
      {
        "content": [
          {
            "data": {},
            "marks": [
              {
                "type": "bold",
              },
              {
                "type": "italic",
              },
            ],
            "nodeType": "text",
            "value": "Administrator cmentarza",
          },
        ],
        "data": {},
        "nodeType": "paragraph",
      },
    ],
    "data": {},
    "nodeType": "document",
  },
  "createdAt": "2024-08-31T00:00+02:00",
  "pathname": "regulamin-cmentarza",
  "title": "Regulamin cmentarza",
}
`;

exports[`contentful integration should return post previews 1`] = `
[
  {
    "createdAt": "2024-05-27T00:00+02:00",
    "pathname": "premiery-we-wtelnie",
    "title": "Premiery we Wtelnie",
  },
]
`;
